package/package.py

```diff
@@ -338,14 +338,26 @@ def build(args: argparse.Namespace) -> str:
 
     package_dir = build_package_dir(metadata, binary_path, attribution_path)
 
+    def infer_package_type(extension: str) -> Tuple[str, str]:
+        package_types = {
+            "rpm",
+            "deb",
+            "tar.gz",
+        }
+        for pkg_type in package_types:
+            if extension.endswith(pkg_type):
+                return pkg_type
+        raise Exception(f"unable to infer package type from extension {ext}")
+
     artifacts = []
-    if not args.no_rpm:
-        artifacts.append(build_rpm(metadata, package_dir))
-    if not args.no_suse_rpm:
-        artifacts.append(build_rpm(metadata, package_dir, "suse"))
-    if not args.no_deb:
-        artifacts.append(build_deb(metadata, package_dir))
-    artifacts.append(build_package_archive(metadata, package_dir))
+    for ext in args.pkg_extensions:
+        pkg_type = infer_package_type(ext)
```

Comment:
nit: the `infer_package_type` function seems a bit redundant. We could just have the `if..elif`s below check the extension directly, e.g. `ext.endswith("rpm"):`

Reply:
and still fallback on the exception if there is no match.